### PR TITLE
Add `updatePriorities` for security app update reminders as a Firebase remote config

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.shortcut.com/simpledotorg/story/8037/add-updatepriority-for-security-app-update-reminders-as-a-firebase-remote-config

### CODE REVIEW

**Comment by msasikanth** [comment]
Since Honey is on leave, I am picking this up.

File: app/src/main/java/org/simple/clinic/main/TypedMap.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.simple.clinic.main
+
+import javax.inject.Qualifier
+
+@Qualifier
+annotation class TypedMap(val type: Type) {
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/main/TypedMap.kt:6]: Nice 👌🏾

File: app/src/main/java/org/simple/clinic/appupdate/CheckAppUpdateAvailability.kt [hunk 2]
```diff
@@ -31,6 +31,10 @@ class CheckAppUpdateAvailability @Inject constructor(
     private val appVersionFetcher: AppVersionFetcher
 ) {
 
+  companion object {
+    const val UPDATE_PRIORITIES_DEFAULT_VALUE = 0
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/appupdate/CheckAppUpdateAvailability.kt:35]: We can shorten the name and call it `DEFAULT_UPDATE_PRIORITY`, wdyt?

File: app/src/main/java/org/simple/clinic/home/patients/PatientsInit.kt [hunk 3]
```diff
@@ -14,7 +14,8 @@ class PatientsInit(
         LoadUser,
         RefreshUserDetails,
         LoadNumberOfPatientsRegistered,
-        LoadInfoForShowingAppUpdateMessage
+        LoadInfoForShowingAppUpdateMessage,
+        LoadAppStaleness
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/home/patients/PatientsInit.kt:18]: I think trying to load app staleness here is causing the tests to fail? 

I am just curious, we are already passing priority in `ShowAppUpdate`, any reason why we didn't do this for staleness as well? Is there any issue with that? 🤔

**Review by msasikanth — CHANGES_REQUESTED** [review]
It looks good overall. Unit tests are failing, can you take a look at that, please?

File: app/src/main/java/org/simple/clinic/home/patients/PatientsInit.kt (hunk 3 above)
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/home/patients/PatientsInit.kt:18]: Not a bad idea. Will do this in the next pr

**Review by msasikanth — APPROVED** [review]

**Pull request merged**